frontend/src/metabase/components/ErrorPages/ErrorDiagnosticModal.tsx

```diff
@@ -0,0 +1,239 @@
+              // eslint-disable-next-line no-literal-metabase-strings -- we're mucking around in the software here
+              label={t`Metabase instance version and settings`}
+            />
+          </Stack>
+          <Alert variant="warning">
+            {t`Review the downloaded file before sharing it, as the diagnostic info may contain sensitive data.`}
+          </Alert>
+          <Flex gap="sm" justify="flex-end" mt="lg">
+            <Button onClick={onClose}>{t`Cancel`}</Button>
+            <FormSubmitButton
+              variant="filled"
+              leftIcon={<Icon name="download" />}
+              label={t`Download`}
+              color="brand"
+            />
+          </Flex>
+        </Form>
+      </FormProvider>
+    </Modal>
+  );
+};
+
+export const ErrorDiagnosticModalTrigger = () => {
+  const { value: errorInfo, loading, error } = useErrorInfo();
+
+  const [isModalOpen, setModalOpen] = useState(false);
+
+  if (loading || !errorInfo) {
+    return (
+      <Center>
+        <Loader />
+      </Center>
+    );
+  }
+
+  if (error) {
+    console.error(error);
+    return null;
+  }
+
+  return (
+    <ErrorBoundary>
+      <Stack justify="center" my="lg">
+        <Box>
+          <Text align="center">
+            {jt`Click the button below to download diagnostic information to send
+            to
+            ${(
+              <Link key="email" variant="brand" to="mailto:help@metabase.com">
+                {t`help@metabase.com`}
+              </Link>
+            )}`}
+          </Text>
+        </Box>
+        <Button
+          leftIcon={<Icon name="download" />}
+          onClick={() => setModalOpen(true)}
+        >
+          {t`Download diagnostic information`}
+        </Button>
+      </Stack>
+      {isModalOpen && (
+        <ErrorDiagnosticModal
+          errorInfo={errorInfo}
+          onClose={() => setModalOpen(false)}
+        />
+      )}
+    </ErrorBoundary>
+  );
+};
+
+export function KeyboardTriggeredErrorModal() {
+  const [
+    isShowingDiagnosticModal,
+    { turnOn: openDiagnosticModal, turnOff: closeDiagnosticModal },
+  ] = useToggle(false);
+
+  useEffect(() => {
+    const keyboardListener = (event: KeyboardEvent) => {
+      if (event.key === "F1" && (event.ctrlKey || event.metaKey)) {
+        openDiagnosticModal();
+      }
+    };
+    window.addEventListener("keydown", keyboardListener);
+    return () => {
+      window.removeEventListener("keydown", keyboardListener);
+    };
+  }, [openDiagnosticModal]);
+
+  const {
+    value: errorInfo,
+    loading,
+    error,
+  } = useErrorInfo({ enabled: isShowingDiagnosticModal });
+
+  if (!isShowingDiagnosticModal || error) {
+    return null;
+  }
+
+  if (loading || !errorInfo) {
+    return (
+      <Modal opened onClose={closeDiagnosticModal}>
+        <Stack align="center" justify="center" mb="lg">
+          <Text w="bold" color="text-medium" mb="sm">
+            {t`Gathering diagnostic information...`}
```

Comment:
You might think about giving a little context here to help translators. For all I know, in one of the languages we support, "diagnostic" can be translated in various ways depending on nuances of the context. "Gathering" might also get mistranslated without some more clues.